x/twap/api_test.go

```diff
@@ -26,63 +30,84 @@ var (
 	tPlus10sp5Record = newTwapRecordWithDefaults(
 		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
 
+	tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
+
 	// accumulators updated from tPlus10sp5Record with
 	// t = baseTime + 20
 	// sp0 = 2, sp1 = 0.5
 	tPlus20sp2Record = newTwapRecordWithDefaults(
 		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
+
+	tPlus20sp2TapRecordAB, tPlus20sp2TapRecordAC, tPlus20sp2TapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
 )
 
 func (s *TestSuite) TestGetBeginBlockAccumulatorRecord() {
 	poolId, denomA, denomB := s.setupDefaultPool()
 	initStartRecord := newRecord(poolId, s.Ctx.BlockTime(), sdk.OneDec(), sdk.ZeroDec(), sdk.ZeroDec())
-	initStartRecord.PoolId, initStartRecord.Height = poolId, s.Ctx.BlockHeight()
-	initStartRecord.Asset0Denom, initStartRecord.Asset1Denom = denomB, denomA
-
+	initStartRecord[0].Height = s.Ctx.BlockHeight()
 	zeroAccumTenPoint1Record := recordWithUpdatedSpotPrice(initStartRecord, sdk.NewDec(10), sdk.NewDecWithPrec(1, 1))
 
+	tapPoolId, tapDenomA, tapDenomB, tapDenomC := s.setupDefaultThreeAssetPool()
+	tapInitStartRecord := newTapRecord(tapPoolId, s.Ctx.BlockTime(), sdk.OneDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	for i := range tapInitStartRecord {
+		tapInitStartRecord[i].Height = s.Ctx.BlockHeight()
+	}
+	tapZeroAccumTenPoint1Record := recordWithUpdatedSpotPrice(tapInitStartRecord, sdk.NewDec(10), sdk.NewDecWithPrec(1, 1))
+
 	tests := map[string]struct {
 		// if start record is blank, don't do any sets
-		startRecord types.TwapRecord
+		startRecord []types.TwapRecord
 		// We set it to have the updated time
-		expRecord  types.TwapRecord
+		expRecord  []types.TwapRecord
 		time       time.Time
 		poolId     uint64
-		quoteDenom string
-		baseDenom  string
+		quoteDenom []string
+		baseDenom  []string
```

Comment:
Agreed, addressed this with your suggestion! 

Also your comment got me to realize that this is a misnomer; this test case does not take quote and base denoms (where order matters), rather it simply takes denom0 and denom1 (where order technically matters but in reality it doesn't because it gets auto lexicographically reordered). Thanks for drawing my attention to it!